fix: chip style

sangameshsomawar opened the pull request:
Fixes: https://github.com/callstack/react-native-paper/issues/3916

Chip component was having `"flexGrow": 1,` style because of which it was taking the entire parent size when a parent has aligned item style. 

### Summary

I have removed `flexGrow`. 

### Test plan

I simulated and tested the example given in the [issue](https://github.com/callstack/react-native-paper/issues/3916)

lukewalczak commented:
Hey @sangameshsomawar, I've checked your changes against the example code in the snack and looks like the issue still exists.

sangameshsomawar commented:
@lukewalczak Hey, I tested it again, and it's working for me! I've even gone ahead and pushed the example code to a [separate branch](https://github.com/sangameshsomawar/react-native-paper/tree/fix/chip-style-testing). Could you please take a look once again? Let me know if there are any test cases that I might have missed. 
 
<img width="558" alt="Screenshot 2023-07-24 at 10 06 28 AM" src="https://github.com/callstack/react-native-paper/assets/93119254/2886e218-cf73-44bc-816f-537a81a0f894">

lukewalczak commented on src/components/Chip/Chip.tsx:
```diff
@@ -473,9 +472,6 @@ const styles = StyleSheet.create({
     alignItems: 'center',
   },
-  touchable: {
-    flexGrow: 1,
```
We should replace then `flexGrow: 1` with `width: '100%'` otherwise we will face the issue with full width `Chip` on the web: https://github.com/callstack/react-native-paper/issues/2804

sangameshsomawar commented:
@lukewalczak Thanks for pointing out the issue. I have fixed it and also added that scenario to 
[example branch](https://github.com/sangameshsomawar/react-native-paper/tree/fix/chip-style-testing)

Screenshots:

<img width="551" alt="Screenshot 2023-07-26 at 10 12 02 PM" src="https://github.com/callstack/react-native-paper/assets/93119254/436b3e3e-040f-40f3-8b10-957ac01d77e3">
<img width="558" alt="Screenshot 2023-07-26 at 10 10 57 PM" src="https://github.com/callstack/react-native-paper/assets/93119254/d55acf6f-a8bf-45d7-ac10-87392c0fff15">
<img width="1840" alt="Screenshot 2023-07-26 at 10 09 17 PM" src="https://github.com/callstack/react-native-paper/assets/93119254/dc2ac420-15b6-405d-a567-4936b0a33617">
ll the screenshots:

sangameshsomawar commented:
> @sangameshsomawar : Have you tried on android on a physical device?

@Angelk90 I could not test in android physical device but I have tested in simulator and posted screenshot in above comment.

lukewalczak commented:
Thanks @sangamesh1439 !
